clients/drcachesim/tracer/tracer.cpp

```diff
@@ -152,6 +154,14 @@ static volatile bool exited_process;
 static bool have_phys;
 static physaddr_t physaddr;
 
+/* priority of registering callbacks for events
+ * using priority = DRMGR_PRIORITY_INSERT_DRWRAP + 1 to make sure the drwrap
+ * registration for function pre/post callbacks happens before memtrace's
+ * meta instruction
```

Comment:
Oh, this is to avoid function trace entries mangling with memory accesses in one single BB.
If we don't specify this, the function trace entries might be appended to the place after the first PC entry for a BB. That will mess up the later memory access entries as those entries are appended according to offset. Always inserting the load-buffer-ptr meta instruction can fix this problem but it will push more compliexity to the later raw2trace parser (entries of memory accesses of a BB is no longer consecutive because of the function trace entries). So I guess it is better to keep the function entries always appended before the first instruction of a BB. Do you think this make senses? If so, I will add the explanation into comments in next commit.

Reply:
Done